### PR TITLE
Variable loop

### PR DESCRIPTION
**Pull request by QinomeD**
Added variable delta loop, added a getter for GameLoopExecutor (so you can use deltaTime in pipelines)
*that should be alright*

### CODE REVIEW

File: src/main/java/net/pod/peaengine/gameloop/GameLoop.java [hunk 1]
```diff
@@ -20,6 +20,19 @@ public static void setLoopType(Class<? extends GameLoopExecutor> type) {
...
     }
 
+    /**
+     * Gets current running {@link GameLoopExecutor}
+     * @throws NullPointerException if executor is null
+     * @return current running {@link GameLoopExecutor}
+     */
+    public static GameLoopExecutor getExecutor() {
+        if (executor == null) {
+            throw new NullPointerException("There is no executor running!");
```
Comment (pod1425) [inline src/main/java/net/pod/peaengine/gameloop/GameLoop.java]: The exception used is not the correct type, should be IllegalStateException instead

File: src/main/java/net/pod/peaengine/gameloop/VariableDeltaLoop.java
Comment (pod1425) [inline src/main/java/net/pod/peaengine/gameloop/VariableDeltaLoop.java:1]: DeltaTimeLoop is more suitable name, please rename

File: src/main/java/net/pod/peaengine/gameloop/GameLoop.java [hunk 2]
```diff
@@ -20,6 +20,19 @@ public static void setLoopType(Class<? extends GameLoopExecutor> type) {
         GameLoop.type = type;
     }
 
+    /**
+     * Gets current running {@link GameLoopExecutor}
+     * @throws NullPointerException if executor is null
+     * @return current running {@link GameLoopExecutor}
+     */
+    public static GameLoopExecutor getExecutor() {
```
Comment (pod1425) [inline src/main/java/net/pod/peaengine/gameloop/GameLoop.java]: The getter for the Thread instance itself poses not quite a "security risk", but is not fool-proof. Please make this method return long instead

**Review by pod1425 — CHANGES_REQUESTED** [review]
Changes needed

File: src/main/java/net/pod/peaengine/gameloop/GameLoop.java (hunk 1 above)
Comment (QinomeD) [reply src/main/java/net/pod/peaengine/gameloop/GameLoop.java]: Oh oop

**Review by pod1425 — APPROVED** [review]

**Pull request merged**